<feat WIP>: augmenting mmlu

SemyonEpanov opened the pull request.

fxlrnrpt commented on src/core/prompts/mmlu_branches_aug.py:
```diff
@@ -0,0 +1,39 @@
+            Return JSON only with:
+            {{"answer":"{letters}","rationale":"1-3 sentences (concise)","key_steps":["fact1","fact2","fact3"]}}
+
+            Answer the MCQ briefly and factually (no step-by-step reasoning).
```
Why? I thought we wanted to elicit step-by-step reasoning

SemyonEpanov replied:
i mean

```
            thinking_config=types.ThinkingConfig(thinking_budget=0)
```

fxlrnrpt replied:
I guess we need to better align on the goal of the experiment then first. Could you add a design doc to `docs` with: hypothesis, execution plan, expected resutls

SemyonEpanov commented on src/core/distillation/synth-aug-mmlu.py:
```diff
@@ -0,0 +1,164 @@
+        },
+    }
+
+def process_tsv(tsv_path, out_jsonl, limit=None):
```
Yes, we can, but, I think the CLI call is more useful.

SemyonEpanov commented:
I accidentally renamed the file and made changes in one commit (sorry about that).

In general: I changed gemini to openrouter, changed the logic to step-by-step reasoning, and left key_steps as the summation.

In the future, I want to merge branch a into branch c, since the first part of c duplicates a

fxlrnrpt commented on docs/explain-mmlu.md:
```diff
@@ -0,0 +1,17 @@
+1) **Main point**
+
+Obtain a synthetic dataset (answers + brief explanations + analysis of erroneous answers + CoT tokens) for training subsequent models.
```
We actually want to elicit the full reasoning chain, don't we?

Could you also add why we want to do it? AFAIU, we want to fine-tune small models on different versions of the distilled CoT and compare the performance. Right?

fxlrnrpt commented on src/core/distillation/synth_aug_mmlu.py:
```diff
@@ -0,0 +1,208 @@
+CHUNK_SIZE = int(os.getenv("SYNTH_CHUNK_SIZE", "16"))
+DUMP_EVERY = int(os.getenv("SYNTH_DUMP_EVERY", "10"))
+
+ALL_LETTERS = [chr(c) for c in range(ord("A"), ord("Z")+1)]
```
Re-use what we have in https://github.com/LabARSS/reasoning-fine-tune/blob/85cc151cdfcac6a5ec409a9f2583486318fe7ed0/src/reasoning_fine_tune/prompts/mmlu_single_token_answer.py#L34? 
Extract it in a separate file for better readibility?

fxlrnrpt replied:
@SemyonEpanov we alrady have `option_ids`

fxlrnrpt commented on src/core/prompts/mmlu_branches_aug.py:
```diff
@@ -0,0 +1,47 @@
+        Return JSON only:
+        {{"correct_answer":"{letters}",
+        "why_correct": "step-by-step reasoning showing why the gold option is correct",
+        "distractor_analysis": {distractor_tpl} }}
```
Could you help me understand `distractor_analysis` vs `why_correct`?

SemyonEpanov replied:
Distractor_analysis explains all answer options. Why_correct explains the correct answer

SemyonEpanov commented on src/core/distillation/synth_aug_mmlu.py:
```diff
@@ -0,0 +1,208 @@
+        record_in = _build_record_in(row_dict, question, choices, letters, gold, model)
+        jobs.append((row.Index, "A", {"question": question, "choices": choices, "gold": gold, "record_in": record_in, "letters": letters}))
+        jobs.append((row.Index, "B", {"question": question, "choices": choices, "gold": gold, "record_in": record_in, "letters": letters}))
+        jobs.append((row.Index, "C", {"question": question, "choices": choices, "gold": gold, "record_in": record_in, "letters": letters}))
```
Yes, that's right. In the new version, it's refactored

fxlrnrpt commented on src/experiments/distill/experiment_aug_mmlu_v0.py:
```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
```
Could we have a dedicated file for each one of the models we want to experiement with?  Like `experiment_aug_mmlu_qwen3_235B.py`
Could you also compile a list of SOTA models we want to run this with?
